data-processing/common/parse_equation.py

```diff
@@ -51,6 +51,8 @@ def get_characters(mathml: str) -> List[Character]:
 def get_symbols(mathml: str) -> List[Symbol]:
     soup = BeautifulSoup(mathml, "lxml")
     node_symbols = []
+    consecutive_chars = []
+    all_indices = []
     for symbol_node in soup.find_all(SYMBOL_TAGS):
```

Comment:
@zkirby does BeautifulSoup guarantee that nodes will be visited in order? If so, can you write a comment confirming that here? If not, this loop may need to be rewritten so that we can make sure that nodes are visited in the left-to-right order that the code below assumes they are for merging mi's.

Reply:
I believe it does, but to be safe I'll change it to `find_all_next` which does.